### PR TITLE
added node.js week one homewok

### PR DESCRIPTION
**Pull request by STEP0HEN**
still working on showing the reviews instead of an empty array for all meals

### CODE REVIEW

File: nodejs/week1/Node-week1-Homework/meal-sharing.app/src/backend/routes/meal.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+const meals = require("../data/meals.json");
+const reviews = require("../data/reviews.json");
+
+
+router.get("/meal", (request, response) => {
+	meals.forEach((meal) => {
+		meal.reviews = reviews.filter((foodreview) => foodreview.mealId === meal.id);
+		return meal
+	});
+	response.json(meals[Math.floor(meals.length * Math.random())]);
```
Comment (OlkaA) [inline nodejs/week1/Node-week1-Homework/meal-sharing.app/src/backend/routes/meal.js]: It is not an error!! But maybe it is a good idea to put meals[Math.floor(meals.length * Math.random())] in some variable and send it in response.json for better code readability.

File: nodejs/week1/Node-week1-Homework/meal-sharing.app/src/backend/routes/large-meals.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const express = require("express");
+const router = express.Router();
+
+const meals = require("../data/meals.json");
+const reviews = require("../data/reviews.json");
+
+
+router.get("/large-meals", (request, response) => {
+	const largeMeals = meals.filter((meal) => meal.price > 80)
```
Comment (OlkaA) [inline nodejs/week1/Node-week1-Homework/meal-sharing.app/src/backend/routes/large-meals.js]: Large meal should be filtered by max number of guests, not a price.

File: nodejs/week1/Node-week1-Homework/warmup/warmup.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	}
+	getCircumference() {
+		const circumference = Math.PI * this.radius * 2;
+		console.log(
+			`The circumference of the circle is ${circumference.toFixed(3)}`
+		);
+	}
+
+	getArea() {
+		const area = Math.PI * (this.radius * this.radius);
```
Comment (OlkaA) [inline nodejs/week1/Node-week1-Homework/warmup/warmup.js]: It is a good idea to change `this.radius * this.radius `by `Math.pow(this.radius, 2)` (to not repeat yourself)
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/pow

**Review by OlkaA — CHANGES_REQUESTED** [review]
Works very well, added some small comments.
I really like that your code is formatted properly and that you added more meals to your json 👍

**Comment by STEP0HEN** [comment]
Thanks so much for the review @OlkaA . I have implemented the changes

File: nodejs/week1/Node-week1-Homework/warmup/warmup.js (hunk 3 above)
Comment (STEP0HEN) [reply nodejs/week1/Node-week1-Homework/warmup/warmup.js]: Okay, thanks for the link

File: nodejs/week1/Node-week1-Homework/meal-sharing.app/src/backend/routes/large-meals.js (hunk 2 above)
Comment (STEP0HEN) [reply nodejs/week1/Node-week1-Homework/meal-sharing.app/src/backend/routes/large-meals.js]: I made the corrections.Thanks

File: nodejs/week1/Node-week1-Homework/meal-sharing.app/src/backend/routes/meal.js (hunk 1 above)
Comment (STEP0HEN) [reply nodejs/week1/Node-week1-Homework/meal-sharing.app/src/backend/routes/meal.js]: ok

**Review by OlkaA — APPROVED** [review]

**Pull request merged**